ledger/onlineaccountscache.go

```diff
@@ -20,16 +20,24 @@ import (
 	"github.com/algorand/go-algorand/data/basics"
 )
 
-const onlineAccountsCacheMaxSize = 1000
+// Worst case memory usage = 2500 * 320 * 150B = 120MB
+const onlineAccountsCacheMaxSize = 2500
 
 type onlineAccountsCache struct {
 	accounts map[basics.Address]*persistedOnlineAccountDataList
 }
 
 // init initializes the onlineAccountsCache for use.
 // thread locking semantics : write lock
-func (o *onlineAccountsCache) init() {
+func (o *onlineAccountsCache) init(accts []persistedOnlineAccountData) {
 	o.accounts = make(map[basics.Address]*persistedOnlineAccountDataList)
+	for _, acct := range accts {
```

Comment:
nit: when initializing, all accounts are ordered, and if the cache becomes full no need to continue iterating. consider optimizing it (or add a comment/TODO)